R/pkg/R/install.R

```diff
@@ -0,0 +1,84 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Functions to install Spark in case the user directly downloads SparkR
+# from CRAN.
+
+#' Download and Install Spark to Local Directory
+#' 
+#' \code{install_spark} downloads and installs Spark to local directory if
+#' it is not found. The Spark version we use is 2.0.0 (preview). Users can
+#' specify a desired Hadoop version, the remote site, and the directory where
+#' the package is installed locally.
+#'
+#' @param hadoop_version Version of Hadoop to install. 2.3, 2.4, 2.6,
+#'        and 2.7 (default)
+#' @param url the base URL of the repositories to use
+#' @param local_dir local directory that Spark is installed to
+#' @rdname install_spark
+#' @name install_spark
+#' @export
+#' @examples
+#'\dontrun{
+#' install_spark()
+#'}
+#' @note install_spark since 2.1.0
+install_spark <- function(hadoop_version = NULL, url = NULL, local_dir = NULL) {
+  version <- paste0("spark-", spark_version_default())
+  hadoop_version <- hadoop_version_default()
```

Comment:
I think we can do 3 things for the mirror url:
(a) have a spark mirror url as an argument to this function so that organizations can use a custom url if they want to use their Spark fork or their internal mirror etc.
(b) have a remote list of mirrors (through say apache) that we use if the argument is NULL
(c) fall back to some hardcoded mirror in the csv file in case the fetch in (b) fails

Regd using the CSV file, I think the biggest pain point is that it will require one more update on each release and we might forget to do this. So using some existing pointer like the latest spark version in the website might be better (not sure if we have something easy to parse ?)